Update OpenAPI improvements (#1095)
ballerina-platform/ballerina-release: release-notes/swan-lake-beta3-release-note.md

- Improved mapping different status code responses in the resource function
- Enhanced generating an OpenAPI schema with Ballerina `typeInclusion` scenarios
- Added resource function API documentation mapping to the OAS description and summary

> - Improved resource function request payload mapping with OAS requestBody

Reviewer: ```suggestion
- Improved the resource function request payload mapping with the OAS `requestBody`
```

- Added the resource signature `http:Cache` annotation mapping to response headers

> - Improved reference resolve for accessing separate module data type to map OAS object schemas

Reviewer: ```suggestion
- Improved reference resolving for accessing a separate module data type to map OAS object schemas
```

> - Improved nullable record field mapping to OAS schema property

Reviewer: ```suggestion
- Improved nullable record field mapping to an OAS schema property
```
